src/DurableTask.Netherite/StorageLayer/Faster/StoreWorker.cs

```diff
@@ -281,6 +282,50 @@ internal enum CheckpointTrigger
             Idle
         }
 
+        void LogCheckpointStats()
+        {
+            long inputQueuePositionLag = this.GetInputQueuePositionLag();
+            this.CheckpointDue(out CheckpointTrigger trigger, out long? compactUntil);
+            
+            // since this is a pure function, we declare it as local static for improved performance
+            static string ReportNullableTaskStatus(Task? t)
+            {
+                if (t == null)
+                {
+                    return "is null";
+                }
+                else
+                {
+                    return t.IsCompleted ? "is completed" : "is not completed";
+                }
+            };
+
+            // since the statistics log is only emitted if the trace level is at least "Debug",
+            // we defer the construction of the string to relieve GC pressure
+            string ConstructLogString() {
+
```

Comment:
nit: in C#, the opening `{` is expected to be on its own line.

Reply:
Yeah, I should have the muscle memory to remember this by now, thanks for the reminder. Applied on my next commit.